File: external/java-impl/src/main/java/armyant/hgoe/indisk/edges/ContainedInEdge.java
package armyant.hgoe.indisk.edges;

import org.hypergraphdb.HGHandle;

/**
 * Created by jldevezas on 2017-10-25.
 */
public class ContainedInEdge extends Edge {
    public ContainedInEdge() {
    }

    public ContainedInEdge(HGHandle... outgoingSet) {
        super(outgoingSet);
    }

    public ContainedInEdge(int tailIndex, HGHandle... targets) {
        super(tailIndex, targets);
    }

    public ContainedInEdge(HGHandle[] head, HGHandle[] tail) {
        super(head, tail);
    }
}

File: external/cpp-impl/src/hgoe/edges/contained_in_edge.cpp
//
// Created by jldevezas on 3/8/18.
//

#include <hgoe/edges/contained_in_edge.h>

#include <utility>

BOOST_CLASS_EXPORT_IMPLEMENT(ContainedInEdge)

ContainedInEdge::ContainedInEdge() = default;

ContainedInEdge::ContainedInEdge(NodeSet tail, NodeSet head) : Edge(boost::move(tail), boost::move(head)) {}

Edge::EdgeLabel ContainedInEdge::label() const {
    return EdgeLabel::CONTAINED_IN;
}
